scripts/dependency_install_functions.sh

```diff
@@ -177,14 +179,10 @@ install_libsodium() {
       # sudo required because of egg file
       "$SUDO_CMD" rm -rf "${PROJECT_ROOT}/external/libsodium"
     fi
-    # Here we are using clone instead of submodule update, because submodule
-    # requires the .git folder exist and the current folder be considered a repo
-    # this creates problems in docker because each time a commit is made the 
-    # .git folder contents are changed causing a fresh rebuild of all containers
-    git clone https://github.com/jedisct1/libsodium.git "${PROJECT_ROOT}/external/libsodium"
-    cd "${PROJECT_ROOT}/external/libsodium"
-    git checkout "$DATAFED_LIBSODIUM_VERSION"
-    ./autogen.sh
+
```

Comment:
```suggestion
    # Official documentation for libsodium indicates this is the preferred way to build libsodium.
    # Using the git repo directly results in build instability because of additional network calls when running
    # autogen.sh.
```